include/ceed.h

```diff
@@ -54,6 +54,9 @@
 #else
 #  define CEED_EXTERN extern
 #endif
+#ifndef CEED_QFUNCTION
```

Comment:
Comment copied over:

I think I'd change this to @jedbrown's suggestion of
```
#define CEED_QFUNCTION(name) \
  static const char name ## _loc[] = __FILE__ ":" #name;        \
  static int name
```

so that we can use 
```
  CeedQFunctionCreateInterior(ceed, 1, mass, mass_loc, &qf_mass);
```

This lets us have qfunction files name differently than the base file without my crazy machinations it the PETSc unified BP code.